TTS/vocoder/configs/modified_hifigan.json

```diff
@@ -0,0 +1,159 @@
+        "url": "tcp:\/\/localhost:54324"
+    },
+
+    // MODEL PARAMETERS
+    "use_pqmf": false,
+
+    // LOSS PARAMETERS
+    "use_stft_loss": false,
+    "use_subband_stft_loss": false,
+    "use_mse_gan_loss": true,
+    "use_hinge_gan_loss": false,
+    "use_feat_match_loss": true,  // use only with melgan discriminators
+    "use_l1_spec_loss": true,
+
+    // loss weights
+    "stft_loss_weight": 0,
+    "subband_stft_loss_weight": 0,
+    "mse_G_loss_weight": 1,
+    "hinge_G_loss_weight": 0,
+    "feat_match_loss_weight": 10,
+    "l1_spec_loss_weight": 45,
+
+    // multiscale stft loss parameters
+    // "stft_loss_params": {
+    //     "n_ffts": [1024, 2048, 512],
+    //     "hop_lengths": [120, 240, 50],
+    //     "win_lengths": [600, 1200, 240]
+    // },
+
+    "l1_spec_loss_params": {
+        "use_mel": true,
+        "sample_rate": 16000,
+        "n_fft": 1024,
+        "hop_length": 256,
+        "win_length": 1024,
+        "n_mels": 80,
+        "mel_fmin": 0.0,
+        "mel_fmax": null
+    },
+
+    "target_loss": "avg_G_loss",  // loss value to pick the best model to save after each epoch
+
+    // DISCRIMINATOR
+    "discriminator_model": "hifigan_discriminator",
+    //"discriminator_model_params":{
+    //    "peroids": [2, 3, 5, 7, 11],
+    //    "base_channels": 16,
+    //    "max_channels":512,
+    //    "downsample_factors":[4, 4, 4]
+    //},
+    "steps_to_start_discriminator": 0,      // steps required to start GAN trainining.1
+    "diff_samples_for_G_and_D": false,   // draw a new sample from the dataset for the D pass.
+
+    // GENERATOR
+    "generator_model": "hifigan_generator",
+    "generator_model_params": {
+        "upsample_factors":[8,8,2,2],
+        "upsample_kernel_sizes": [16,16,4,4],
+        "upsample_initial_channel": 512,
+        "resblock_kernel_sizes": [3,7,11],
+        "resblock_dilation_sizes": [[1,3,5], [1,3,5], [1,3,5]],
+        "resblock_type": "1"
+    },
+
+    // DATASET
+    "data_path": "/home/erogol/gdrive/Datasets/non-binary-voice-files/vo_voice_quality_transformation/",
+    "feature_path": null,
+    // "feature_path": "/home/erogol/gdrive/Datasets/non-binary-voice-files/tacotron-DCA/",
+    "seq_len": 8192,
+    "pad_short": 2000,
+    "conv_pad": 0,
+    "use_noise_augment": false,
+    "use_cache": true,
+    "reinit_layers": [],    // give a list of layer names to restore from the given checkpoint. If not defined, it reloads all heuristically matching layers.
+
+    // TRAINING
+    "batch_size": 16,       // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.
+
+    // VALIDATION
+    "run_eval": true,
+    "test_delay_epochs": 10,  //Until attention is aligned, testing only wastes computation time.
+    "test_sentences_file": null,  // set a file to load sentences to be used for testing. If it is null then we use default english sentences.
+
+    // OPTIMIZER
```

Comment:
Doesn't `optimizer_gen = getattr(torch.optim, c.optimizer)` required `optimizer` key in config file or it load any default optimizer?

Reply:
it should be fixed with the new commits.